lib/elixir/lib/exception.ex

```diff
@@ -626,6 +626,8 @@ defmodule UndefinedFunctionError do
         "undefined function"
       not is_nil(module) and :code.is_loaded(module) === false ->
         message(%{e | reason: :"module could not be loaded"})
+      not is_nil(module) and macro_with_same_mfa?(module, function, arity) ->
```

Comment:
I think we don't need to introduce new `:"macro with same mfa"` reason.
Other reasons here are "native" and exist in Erlang.
We can rewrite the error in the clause for `:"function not exported"` reason, and also we have exports available there.